### PR TITLE
Most recent photo

### PR DESCRIPTION
**Pull request by caachz**
### Type of change made:
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling/UX
### Detailed Description
When you create a new journal entry for a plant the URL field for that plant will be updated. 
Also when you create a plant it will create a journal entry for the first photo to tie that to the plant. 
### What Issue does this fix?
closes issue 24
### Why is this change required? What problem does it solve?
The FE team wanted to be able to have the most recent photo related to the plant easily accessible.
### Were there any challenges that arose while implementing this feature? If so, how were the addressed?
Jonathan realized we needed to do the journal entry creation for the first plant photo so we don't lose that info. 
### Components/Files modified:
app.py

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def get_user_plants(user_id):
 def get_one_plant(user_id, plant_id):
     user = mongo.db.users.find_one_or_404({'_id': user_id})
     plant = mongo.db.plants.find_one_or_404({'_id': plant_id})
+    photo = mongo.db.journal.find_one({'user_id': user_id})
```
Comment (jonathan-tschida) [inline app.py]: Did you mean to have this line here? It doesn't seem necessary to get the journal entry if we aren't using it

File: app.py [hunk 2]
```diff
@@ -68,6 +69,15 @@ def create_plant(user_id):
         'care': request.json.get('care', {})
     }
     mongo.db.plants.insert_one(plant)
+
+    journal_entry = {
+        '_id': str(ObjectId()),
+        'plant_id': plant['_id'],
+        'entry_type': 'image',
+        'timestamp': timestamp,
+        'info': plant['image_url'],
```
Comment (jonathan-tschida) [inline app.py]: This needs to be nested inside an object to match with our schema
```
'info': {
    'imageURL': plant['image_url'],
    'notes': 'Plant added to garden!'
}
```

File: app.py [hunk 3]
```diff
@@ -106,6 +116,10 @@ def create_journal_entry(user_id, plant_id):
     user = mongo.db.users.find_one_or_404({'_id': user_id})
     plant = mongo.db.plants.find_one_or_404({'_id': plant_id})
     timestamp = int(datetime.now().timestamp() * 1000)
+
+    if request.json.get('entryType') == 'image':
+        mongo.db.plants.update_one({'_id': plant_id}, {'$set': {'image_url': request.json.get('info')}})
+
```
Comment (jonathan-tschida) [inline app.py]: I think this is also not going to work with our current schema, info is a dict with properties of 'imageURL' and 'notes' we can look into un-nesting these if it is going to be a problem

**Review by jonathan-tschida — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (caachz) [reply app.py]: Good call :D I was trying a different approach first and this was left over from that. I removed it.

File: app.py (hunk 2 above)
Comment (caachz) [reply app.py]: I updated it to account for the nested URL :D

File: app.py (hunk 3 above)
Comment (caachz) [reply app.py]: No need to look at un-nesting it. I changed it to account for nesting.

**Pull request merged**